sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -100,7 +100,13 @@
                 // is not possible in reactor-netty to do this without copying occurring within that library. This
                 // issue has been reported to the reactor-netty team at
                 // https://github.com/reactor/reactor-netty/issues/1479
-                hdr.getValuesList().forEach(value -> reactorNettyRequest.addHeader(hdr.getName(), value));
+                if (hdr.getName().equalsIgnoreCase("User-Agent")) {
+                    // The Reactor-Netty request headers include a User-Agent by default.
+                    // We need to replace this completely as multiple User-Agents is invalid.
+                    reactorNettyRequest.header(hdr.getName(), hdr.getValue());
```

Comment:
Instead of hardcoding this for a specific header, can we instead look at all headers in `restRequest.getHeaders()` and replace all headers in reactorNettyRequest.headers() that have the same name? I just worry about netty adding more headers in the future that will result in duplicates.

Reply:
This is a fantastic idea, I will make the change once the live tests I triggered complete